Shared/sdk/SharedUtil.Hash.hpp

```diff
@@ -818,5 +818,22 @@ namespace SharedUtil
 
         out->assign((char*)buffer, numPasses * 4);
         delete[] buffer;
+
+        if (trimPaddingAtEnd)
+        {
+            // Check last block for 0 padding, and delete
+
+            auto iter = out->end() - 1;
+            if (*iter != 0)
+                return;
+
+            const auto blockEnd = out->end() - 4;
```

Comment:
This doesn't seem correct. If I encode `abc\0\0\0`, then the decode result (pre-trim) would be `abc\0\0\0\0\0`. After trimming it ends up being `abc\0`, which is incorrect. We should trim the exact amount of excess padding that was added previously..

Reply:
Previous version did the same thing. I think that should be looked at in a new pull request

Reply:
I feel like doing `strlen` on a 10-15 mb string isn't particularly efficient.

Reply:
So it this the correct behavior?

Reply:
@sbx320  I don't see a way to achieve what you want. We can't possible figure out if it was padding, or the user has added the `\0\0\0`'s. Please correct me if there's a good way to do so.

Reply:
I can trim all `0`'s, not just the ones in the last block. That would leave us with a string that has no padding. But if the user added `0` we can't figure out if it was added by him, or by us.